### PR TITLE
add Linked List

### PR DESCRIPTION
**Pull request by carloshmartins**
This PR include a full implementation of Linked List

PS: Even is written in Swift, this implementation is not "Swift-ish", I mean that this implementation does not have Copy-or-Write performance and Swift Protocols, such as: Sequence and Collection for better performance.

### CODE REVIEW

**Comment by carloshmartins** [comment]
@vbrazo Please, would you review my PR?

File: data_structures/Linked List/LinkedList.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      tail = prev
+      return current.value
+    }
+    
+    @discardableResult
+    public mutating func remove(after node: Node<Value>) -> Value? {
+      defer {
+        if node.next === tail {
+    tail = node }
+        node.next = node.next?.next
```
Comment (vbrazo) [inline data_structures/Linked List/LinkedList.swift:110]:
```suggestion
        if node.next === tail {
          tail = node 
        }
        node.next = node.next?.next
```

**Review by vbrazo — CHANGES_REQUESTED** [review]
Could you add some tests for your implementation? At least a few lines calling the classes so that we can reproduce the behavior in a playground.

The algorithms in this repository aren't using `XCTestCase` yet. I'd consider adding this in the long run and also link in the CI.

**Comment by carloshmartins** [comment]
I added Linked List Test Scenario and DS Store into gitignore

**Review by vbrazo — APPROVED** [review]

**Pull request merged**